pkg/controllers/customcluster_controller.go

```diff
@@ -90,13 +92,23 @@ const (
 	CustomClusterScaleDownAction customClusterManageAction = "scale-down"
 	KubesprayScaleDownCMDPrefix  customClusterManageCMD    = KubesprayCMDPrefix + "remove-node.yml -vvv -e skip_confirmation=yes"
 
-	// TODO: support custom this in CustomCluster/CustomMachine
-	DefaultKubesprayImage = "quay.io/kubespray/kubespray:v2.20.0"
+	CustomClusterUpgradeAction customClusterManageAction = "upgrade"
+	KubesprayUpgradeCMDPrefix  customClusterManageCMD    = KubesprayCMDPrefix + "upgrade-cluster.yml -vvv "
 
 	// CustomClusterFinalizer is the finalizer applied to crd.
 	CustomClusterFinalizer = "customcluster.cluster.kurator.dev"
 	// custom configmap finalizer requires at least one slash.
 	CustomClusterConfigMapFinalizer = CustomClusterFinalizer + "/configmap"
+
+	// KubeVersionPrefix is the prefix string of version of kubernetes
+	KubeVersionPrefix = "kube_version: "
+
+	// DefaultKubesprayVersion is the version of Kubespray used by the customCluster manager worker pod.
+	DefaultKubesprayVersion = "v2.20.0"
+	// DefaultMaxVersion is the max version of Kubernetes that is supported by DefaultKubesprayVersion.
+	DefaultMaxVersion = "v1.24.6"
+	// DefaultMinVersion is the min version of Kubernetes that is supported by DefaultKubesprayVersion.
+	DefaultMinVersion = "v1.22.0"
```

Comment:
Can we remove such limitation, in my mind we should support more versions by using different kubespray images  if this is the limitation of kubespray

Not a must in this pr, could you file an issue about it

Reply:
yes, it is possible to remove such limitation. If kubespray support the kubernetes version, we can initialize or upgrade to that version using the corresponding version of Kubespray .  This is a new issue that needs to be addressed #218